server/src/main/java/org/apache/iotdb/db/utils/DeviceIdCache.java

```diff
@@ -0,0 +1,53 @@
+package org.apache.iotdb.db.utils;
+
+import org.apache.iotdb.db.exception.PathErrorException;
+import org.apache.iotdb.db.exception.StorageGroupException;
+import org.apache.iotdb.tsfile.exception.cache.CacheException;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public abstract class DeviceIdCache {
```

Comment:
The number of devices in a system is rather limited and holding all in memory is not a burden, so I do not suggest making it an LRU cache for meaningless overhead. 
Nevertheless, I seriously doubt the performance of using a List, as the `contains()` method can be extremely slow even when there are only thousands of devices (not to mention you are using a COWList). I strongly recommend using a HashSet.

Reply:
I'm confused that whether I should use `mNodeCache `directly because its key is just `deviceId`. However, if I choose to use `mNodeCache`, I will have trouble with handling exceptions since I need `PathErrorException `and `StorageGroupException` to distinguish two different cases while `mNodeCache` throws `CacheException` in any case. Do you have any suggestions?